plasmapy/dispersion/analytical/tests/test_mhd_wave_class.py

```diff
@@ -0,0 +1,159 @@
+import astropy.units.core
+import numpy as np
+import pytest
+
+from astropy import units as u
+
+from plasmapy.dispersion.analytical.mhd_waves_ import mhd_waves
+from plasmapy.particles.exceptions import InvalidIonError
+from plasmapy.utils.exceptions import PhysicsWarning
+
+kwargs_plasma_cold = {
+    "B": 1e-3 * u.T,
+    "density": 1e16 * u.m**-3,
+    "ion": "p+",
+}
+kwargs_wave_limits = {
+    "k": 1e-5 * u.rad / u.m,
+    "theta": [0, 90] * u.deg,
+}
+
+sample_waves = mhd_waves(**kwargs_plasma_cold, T=1e6 * u.K)
+
+
+class TestMHDWave:
+    @pytest.mark.parametrize(
+        ("kwargs", "error"),
+        [
+            ({**kwargs_plasma_cold, "B": "wrong type"}, TypeError),
+            ({**kwargs_plasma_cold, "B": [8e-9, 8.5e-9] * u.T}, ValueError),
+            ({**kwargs_plasma_cold, "B": -1 * u.T}, ValueError),
+            ({**kwargs_plasma_cold, "B": 5 * u.m}, u.UnitTypeError),
+            ({**kwargs_plasma_cold, "ion": {"not": "a particle"}}, TypeError),
+            ({**kwargs_plasma_cold, "ion": "e-"}, InvalidIonError),
+            ({**kwargs_plasma_cold, "ion": "He", "Z": "wrong type"}, TypeError),
+            ({**kwargs_plasma_cold, "density": "wrong type"}, TypeError),
+            ({**kwargs_plasma_cold, "density": [5e6, 6e6] * u.m**-3}, ValueError),
+            ({**kwargs_plasma_cold, "density": -5e6 * u.m**-3}, ValueError),
+            ({**kwargs_plasma_cold, "density": 2 * u.s}, u.UnitTypeError),
+            ({**kwargs_plasma_cold, "T": "wrong type"}, TypeError),
+            ({**kwargs_plasma_cold, "T": [1.4e6, 1.7e6] * u.K}, ValueError),
+            ({**kwargs_plasma_cold, "T": -10 * u.eV}, ValueError),
+            ({**kwargs_plasma_cold, "T": 2 * u.s}, u.UnitTypeError),
+            ({**kwargs_plasma_cold, "gamma": "wrong type"}, TypeError),
+        ],
+    )
+    def test_raises_init(self, kwargs, error):
+        """Test scenarios that raise an `Exception`."""
```

Comment:
```suggestion
        """Test scenarios that raise an exception."""
```
...since the docstrings for tests don't show up in the docs.